### PR TITLE
fail if dependencies are missing

### PR DESCRIPTION
**Pull request by xloem**
moved from https://github.com/libbitcoin/libbitcoin-system/pull/1171

I have no experience with GSL, but this appears to work.

### CODE REVIEW

File: templates/gsl.configure.ac [hunk 1]
```diff
@@ -847,6 +847,10 @@ AS_CASE([${$(my.symbol)}], [yes],
 .   define my.prefix = is_true(my.dependency.unprefixed) ?? "" ? "lib"
 .   define my.symbol = find_option_symbol(my.dependency, my.configure)?
 .   define my.check = "$(my.prefix)$(my.name) >= $(my.version)"
+.   if (defined(my.symbol))
+.       define my.option = my.configure->option(option.name =  my.dependency.option)
```
Comment (evoskuil) [inline templates/gsl.configure.ac:851]: Style: double space after `=`

File: templates/gsl.configure.ac [hunk 2]
```diff
@@ -847,6 +847,10 @@ AS_CASE([${$(my.symbol)}], [yes],
...
+.       define my.opttype = get_option_type(my.option)
```
Comment (evoskuil) [inline templates/gsl.configure.ac:852]: `my.opttype`

Style: avoid abbreviations and don't combine distinct words. For example, use `my.option.type`.

**Review by evoskuil — CHANGES_REQUESTED** [review]
Thanks, sorry for the review delay.

**Comment by xloem** [comment]
Thanks for the review.  I'm afraid I don't have this set up right now to test changes; it could possibly be quite some time.  Feel free to close if it's in the way; otherwise I'll get back to it eventually.

**Comment by evoskuil** [comment]
No worries, I’ll make the edits.

**Comment by evoskuil** [comment]
Added with suggested style changes in new PR. Thanks!

**Comment by xloem** [comment]
Thanks for sorting this out.

**Pull request closed without merging**